app/src/main/AndroidManifest.xml

```diff
@@ -21,6 +21,8 @@
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+
+            <nav-graph android:value="@navigation/splash_graph" />
```

Comment:
Since this is used to determine the `deeplink` configuration, I'd set `authentication_graph` here instead.
Also, I know the docs say that you can keep the default behavior if you don't change the `launchMode` of the activity. But many times we do have to change it to `singleTask`. I believe it would be a good exercise for you to do the same, and override onNewIntent in MainActivity and call the handle deep link method from the docs